### ISSUE TITLE
Report error for nullable annotation on interface type

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#29000]
The compiler should report an error for `IEnumerable?` since annotations can only be applied to value types or reference types.
```c#
using System.Collections;
class C<T> where T : IEnumerable?
{
}
```

**Comment by cston** [comment]
`static void F(IEnumerable? e)` should be allowed since `e` is a boxed value.

**Comment by cston** [comment]
`where T : IEnumerable?` is valid.
